### DISCUSSION TITLE
3.0.0rc2

### DISCUSSION THREAD

**Discussion by michael-s-molina** [apache/superset#24824] [General]
Hello Superset Community,

This is a call for the vote to release Apache Superset version 3.0.0.

The release candidate:
https://dist.apache.org/repos/dist/dev/superset/3.0.0rc2/

Git tag for the release:
https://github.com/apache/superset/tree/3.0.0rc2

The Change Log for the release:
https://github.com/apache/superset/blob/3.0.0rc2/CHANGELOG.md

The Updating instructions for the release:
https://github.com/apache/superset/blob/3.0.0rc2/UPDATING.md

Public keys are available at:
https://www.apache.org/dist/superset/KEYS

The vote will be open for at least 72 hours or until the necessary number
of votes are reached.

Please vote accordingly:

[ ] +1 approve
[ ] +0 no opinion
[ ] -1 disapprove with the reason

Thanks,
The Apache Superset Team

<hr /><em>This discussion was created from the release <a href='https://github.com/apache/superset/releases/tag/3.0.0rc2'>3.0.0rc2</a>.</em>

**Comment by EugeneTorap** [comment]
Hi @michael-s-molina!
We have to fix & merge #24803 before releasing 3.0.0 version of superset because it's an important Py3.11 bug

**Reply by michael-s-molina** [reply]
Thank you for working on a fix @EugeneTorap. If we have more RCs we can include it. If not, it will be included in the next patch. Keep in mind that we'll increase the cadence of patch releases given that we have fixes every day.

**Comment by sqja** [comment]
Hello,

I do confirm that with 3.00 RC2 we managed to reproduce the bug mentioned in the discussion started in 3.0 RC1 thread and that you asked me to test
https://github.com/apache/superset/discussions/24581#discussioncomment-6551385
Link to the bug is https://github.com/apache/superset/issues/21508

Attached screenshot of the bug with 3.00 RC2
![image](https://github.com/apache/superset/assets/130609036/4f258508-69a2-4b39-98dd-b2b3c2753a8e)

**Reply by michael-s-molina** [reply]
Thank you for the verification @sqja. I added a comment to the issue.

**Reply by michael-s-molina** [reply]
Fix: https://github.com/apache/superset/pull/25009

**Comment by john-bodley** [comment]
Issue with native dashboard filter database migration. Fixed by https://github.com/apache/superset/pull/24831.

**Comment by martin-raymond** [comment]
hi,

i found the following issue on the version rc2 : Can't use cross-filtering on a column from a table whose name has been overidden, https://github.com/apache/superset/issues/24836

thx

**Comment by john-bodley** [comment]
Issue with rendering annotation layers. Fixed by https://github.com/apache/superset/pull/24874.

**Comment by lf-floriandin** [comment]
The pip release doesn't seem to work on Debian Systems (Ubuntu 22.04 in this case)
Got an ansible script that deploys Superset on an Ubuntu Server. Works fine for 2.1, but with 3.0.0Rc2 I can't login and get an error: `flask_wtf.csrf.CSRFError: 400 Bad Request: The CSRF session token is missing.`.

**Reply by michael-s-molina** [reply]
@dpgaspar Can this be related to [Flask App Builder](https://github.com/dpgaspar/Flask-AppBuilder)? There's also https://github.com/apache/superset/issues/24717 which seems to be related.

**Reply by michael-s-molina** [reply]
This seems to be fixed by configuring TALISMAN. See https://github.com/apache/superset/issues/24579 for more details.

**Reply by natechoiniere** [reply]
Referencing #24579, I set WTF_CSRF_ENABLED = False and TALISMAN_ENABLED=False with no success. What did work was downgrading to 2.1.0, but then I could not edit my dashboards (couldn't even change the color theme; hitting "Save" did nothing!). This all started with me grabbing a fresh installation of the latest tagged release and then trying to access the page from an external network using the host machine's IP + port 8088, only to realize that attempting to log in would simply refresh the login page. 

At this point I'm stuck using 2.1.0 for general use, and when I want to edit my Dashboard I must `docker-compose down` and change the image to 2.1.0 in my `docker-compose.yml`, and only once I `docker-compose up -d` can I change my Dashboard, before downing/upping back to 2.1.0 for general use. At this point I'm convinced that 3.0.0 is just straight broken my G

**Reply by michael-s-molina** [reply]
@natechoiniere 3.0.0RC2 is not an official release, it's a [release candidate](https://www.tutorialspoint.com/software_testing_dictionary/release_candidate.htm), it may contain many bugs and should never be tested using a production database. 3.0.0RC2 is a major version and it contains many breaking changes and migrations described in `CHANGELOG.md` and `UPDATING.md` files. Make sure you executed the downgrade migrations before going back to 2.1.0. They may be the root cause for the problems you mentioned. Regarding the login errors on 3.0, we're monitoring to see if it's a configuration problem or a hidden bug. So far, the TALISMAN configuration seems to be working for the majority of users so your problem may be related to something specific in your deployment. I suggest opening an issue and trying to describe as much as you can about your environment so folks from the community can help. We also have the `#debug-together` and `#deploying-superset` Slack channels as good places to get help.

**Comment by ahmed-adly-khalil** [comment]
is there a way to fix this in 3.0 https://github.com/apache/superset/issues/24837 ? basically adding a permission to public role takes it away from the admin role

**Reply by michael-s-molina** [reply]
Hi @ahmed-adly-khalil. We can include a fix for this as soon as one is available. If it misses 3.0.0, it can always be included in a patch release 3.0.X.

**Comment by mknadh** [comment]
Hi @michael-s-molina,

Please make sure to have the fix for https://github.com/apache/superset/issues/24579 before releasing 3.0, as this is blocking helm deployments of 3.0.0-rc2 on kubernetes clusters.

**Reply by michael-s-molina** [reply]
@mknadh The issue you linked has many comments pointing to configuring TALISMAN as the solution. If some additional context is missing for Kubernetes clusters, I suggest creating an issue with the details. If the problem persists and someone submits a fix for it, I'll include it in 3.0.0 or 3.0.X depending on when the fix is available.

**Comment by Chaifist** [comment]
Hi,  @michael-s-molina,
  In RC1 requirements holidays==0.23,  in RC2 updated holidays==0.28, 
but in python 3.9.17, prophet==1.1.4. [not work](https://github.com/facebook/prophet/issues/2354) solved "Downgrading to 1.1.1 solves the problem." but prophet==1.1.1 need holidays==0.23 in RC1 solves this problem

**Reply by michael-s-molina** [reply]
Thank you @Chaifist. I submitted a fix [here](https://github.com/apache/superset/pull/25017).

**Comment by zhaorui2022** [comment]
When I tested 3.0.0rc2, I ran into this error #25064

**Reply by michael-s-molina** [reply]
@zhaorui2022 I added a reply to the issue. Let's track there 😉

**Comment by michael-s-molina** [comment]
Closing in favor of the [3.0.0rc3 discussion](https://github.com/apache/superset/discussions/25076).
